### PR TITLE
[AWS] Disable unattended-upgrade with cloud-init

### PR DESCRIPTION
**Pull request by Michaelvll**
This PR is separated from #1641, as the disabling of the unattended upgrade for AWS is more urgent for the AWS user.

We are now using the cloud-init to disable the `unattended-upgrade` which holds the lock for `apt`. The user's `sudo apt install` locked issue should now be fixed and deterministic.

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c test-fm ./examples/using_file_mounts.yaml` (which contains the `sudo apt install tree` in setup)
  - [x] `sky launch -c test-fm1 --num-nodes 1 ./examples/using_file_mounts.yaml`
  - [x] `sky launch -c min` immediately log into the VM after provision and run `sudo apt install tree`
- [x] All smoke tests: `pytest tests/test_smoke.py --aws`

### CODE REVIEW

File: sky/templates/aws-ray.yml.j2 [hunk 1]
```diff
@@ -59,6 +59,18 @@ available_node_types:
           #   SpotOptions:
           #       MaxPrice: MAX_HOURLY_PRICE
       {% endif %}
+      UserData: |
+        #cloud-config
+        users:
+          - name: skypilot:ssh_user
+            shell: /bin/bash
+            sudo: ALL=(ALL) NOPASSWD:ALL
+            ssh-authorized-keys:
```
Comment (concretevitamin) [inline sky/templates/aws-ray.yml.j2:68]: https://cloudinit.readthedocs.io/en/latest/reference/examples.html#including-users-and-groups

Does `ssh-authorized-keys` work? The above link suggests `ssh_authorized_keys`.

File: sky/templates/aws-ray.yml.j2 [hunk 2]
```diff
@@ -59,6 +59,18 @@ available_node_types:
...
+              - skypilot:ssh_public_key_content
+        bootcmd:
```
Comment (concretevitamin) [inline sky/templates/aws-ray.yml.j2:70]: Nice! Maybe add a ref: https://unix.stackexchange.com/a/471192?

File: sky/authentication.py [hunk 3]
```diff
@@ -97,17 +109,7 @@ def setup_aws_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
     # ec2.describe_key_pairs.
     # Note that sudo and shell need to be specified to ensure setup works.
     # Reference: https://cloudinit.readthedocs.io/en/latest/reference/modules.html#users-and-groups  # pylint: disable=line-too-long
```
Comment (concretevitamin) [inline sky/authentication.py:110]: It seems like these few lines should be moved to the .j2 file?

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll! Some comments.

**Review by concretevitamin — APPROVED** [review]
LGTM! modulo a question on restore exceptions.

File: sky/backends/backend_utils.py [hunk 4]
```diff
@@ -127,8 +127,13 @@
 #    it's possible to failover to 1b, which leaves a leaked instance in 1a. Here,
 #    we use the new yaml's zone field, which is guaranteed to be the existing zone
 #    '1a'.
+# - UserData: The UserData field of the old yaml may be outdated, and we want to
+#   use the new yaml's UserData field, which contains the authorized key setup as
+#   well as the disabling of the auto-update with apt-get.
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/backends/backend_utils.py:132]: Yes, I just tried to launch a cluster with the master branch and relaunch it with the current PR. It seems the UserData become the latest version.
For running cluster, the `unattended-upgrade` should be disabled by our previous `sudo kill` stuffs.
For stopped cluster, the `bootcmd` in the current PR will take effect and disable the `unattended-upgrade`

File: sky/authentication.py [hunk 5]
```diff
@@ -106,8 +106,6 @@ def setup_aws_authentication(config: Dict[str, Any]) -> Dict[str, Any]:
     # Use cloud init in UserData to set up the authorized_keys to get
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/authentication.py]: Just removed and renamed the function. : )

File: sky/backends/backend_utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (concretevitamin) [reply sky/backends/backend_utils.py:132]: Awesome!

**Pull request merged**